### ISSUE TITLE
Routing does not work when the base folder and URI component has the same name

### ISSUE THREAD

**Issue by libregeek** [bcosca/fatfree-core#176]
I have an F3 application named *sample* which is installed in a folder named *sample* and one of the modules in this application is also named as  *sample*. 

The application is accessed from http://localhost/sample and the module is accessed from http://localhost/sample/sample. This works fine without any issues in v3.5.1. But it throws a 404 error after I upgraded F3 to v3.6.0. The following is the routes.ini config.

GET /sample=\Sample\Controller->index
POST /sample/send=\Sample\Controller->send
GET /sample/generate=\Sample\Controller->generate

Is this a regression or am I doing something not in F3 standards?

**Comment by ikkez** [comment]
This should work just fine. Do you use the default `.htaccess` file?

**Comment by libregeek** [comment]
Here is the .htaccess I'm using:

```
# Enable rewrite engine and route requests to framework
RewriteEngine On
# Some servers require you to specify the `RewriteBase` directive
# In such cases, it should be the path (relative to the document root)
# containing this .htaccess file
#
# RewriteBase /
RewriteRule ^(tmp)\/|\.ini$ - [R=404]
RewriteCond %{REQUEST_FILENAME} !-l
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule .* index.php [L,QSA]
RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization},L]
```

**Comment by ikkez** [comment]
try to set `RewriteBase /sample/`

**Comment by libregeek** [comment]
I tried to set RewriteBase, but it doesn't work. 

However, when I commented  the following line in base.php it works without any issues. 
https://github.com/bcosca/fatfree-core/blob/master/base.php#L1512

I have also noticed that the application works without any issues if it is moved to another folder. For example, it works if I move the application to a folder named projects and access it using http://localhost/projects/sample/sample

**Comment by geniuswebtools** [comment]
I've had issues like this in the past where the rewritebase didn't work, however, changing the rewrite rule to include the subfolder will sometimes work:

RewriteRule .* /sample/index.php [L,QSA]

**Comment by libregeek** [comment]
@geniuswebtools I'm sorry your suggestion didn't work

@ikkez is this a regression in the new verison? This was working fine in the previous version (3.5.1) 

When I try to access `http://localhost/sample/sample/details/3` The error is as follows:

> Not Found
> HTTP 404 (GET /sample/details/3)
> [sample/vendor/bcosca/fatfree-core/base.php:1631] Base->error()
> [sample/index.php:19] Base->run()

**Comment by kavin-90** [comment]
I checked 3.6.x works superb no problem on Nginx

**Comment by kavin-90** [comment]
I checked 3.6.x works superb no problem on Nginx.

My Routes were
http://localhost/sample/public
http://localhost/sample/public/sample/details/3

**Comment by libregeek** [comment]
@kavin-90 can you try `http://localhost/sample/sample/`

**Comment by kavin-90** [comment]
i tried it works 😄

**Comment by xfra35** [comment]
@libregeek, it should be fixed now.

This bug was introduced in https://github.com/bcosca/fatfree-core/commit/502023dfb15c9b8348eab7aa73019adb3836c8c3 and caused the `BASE` to be stripped twice from the request URI.

This explains why `'/sample/sample'` was processed as `''`.

**Comment by libregeek** [comment]
@xfra35 thanks for fixing it.

**Comment by libregeek** [comment]
@xfra35 is it possible to have a bug fix release for v3.6?

**Comment by ikkez** [comment]
We'll release it this week. @bcosca @xfra35
